### PR TITLE
DF-20045 - Separate LWBA & price endpoints

### PR DESCRIPTION
**Pull request by mjk90**
## Description
A follow up to the LWBA validation changes. This PR separates LWBA & price endpoints in:
- Elwood
- NCFX

and updates GSR to use a custom Adapter implementation, allowing LWBA validation while using the same transport & cache keys for price & LWBA endpoints. This was necessary because GSR doesn't allow multiple WS connections and doesn't seem to handle multiple subscription requests from the same connection well.

......

## Steps to Test

- yarn test:unit
- yarn test:integration

## Quality Assurance

- [x] If a new adapter was made, or an existing one was modified so that its environment variables have changed, update the relevant `infra-k8s` configuration file.
- [x] If a new adapter was made, or an existing one was modified so that its environment variables have changed, update the relevant `adapter-secrets` configuration file or update the [soak testing blacklist](/packages/scripts/src/get-changed-adapters/soakTestBlacklist.ts).
- [x] If a new adapter was made, or a new endpoint was added, update the `test-payload.json` file with relevant requests.
- [x] The branch naming follows git flow (`feature/x`, `chore/x`, `release/x`, `hotfix/x`, `fix/x`) or is created from Jira.
- [x] This is related to a maximum of one Jira story or GitHub issue.
- [x] Types are safe (avoid TypeScript/TSLint features like any and disable, instead use more specific types).
- [x] All code changes have 100% unit and integration test coverage. If testing is not applicable or too difficult to justify doing, the reasoning should be documented explicitly in the PR.

### CODE REVIEW

File: packages/sources/elwood/src/endpoint/crypto.ts [hunk 1]
```diff
@@ -18,27 +14,14 @@ const inputParameters = new InputParameters(priceEndpointInputParametersDefiniti
   },
 ])
 
-type OmitResultFromLwba = Omit<LwbaResponseDataFields, 'Result'>
-
 export type BaseEndpointTypes = {
   Parameters: typeof inputParameters.definition
   Settings: typeof config.settings
-  Response: OmitResultFromLwba & SingleNumberResultResponse
+  Response: SingleNumberResultResponse
```
Comment (mxiao-cll) [inline packages/sources/elwood/src/endpoint/crypto.ts:20]: This is not backward compatible so should be a major version bump?

File: packages/sources/elwood/src/endpoint/crypto-lwba.ts [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+  },
+])
+
+export type BaseEndpointTypesLwba = {
+  Parameters: typeof inputParameters.definition
+  Settings: typeof config.settings
+  Response: LwbaResponseDataFields
+}
+
+export const cryptoLwbaEndpoint = new LwbaEndpoint({
```
Comment (karen-stepanyan) [inline packages/sources/elwood/src/endpoint/crypto-lwba.ts:23]: Since this is a LWBA endpoint, should we have `customOutputValidation` here ?

Comment (karen-stepanyan) [reply packages/sources/elwood/src/endpoint/crypto-lwba.ts:23]: Oh nevermind, LwbaEndpoint already has that internally.

File: packages/sources/elwood/src/transport/lwba.ts [hunk 3]
```diff
@@ -0,0 +1,165 @@
...
+import { WebSocketTransport } from '@chainlink/external-adapter-framework/transports'
+import { makeLogger } from '@chainlink/external-adapter-framework/util'
+import { EndpointContext } from '@chainlink/external-adapter-framework/adapter'
+import axios from 'axios'
+
+const logger = makeLogger('ElwoodWsLwba')
+
+const DEFAULT_TRANSPORT_NAME = 'default_single_transport'
+
+export type SubscribeRequest = {
```
Comment (karen-stepanyan) [inline packages/sources/elwood/src/transport/lwba.ts:11]: It seems that this and bellow types are identical with `crypto` transport as well. Maybe we can move them to a shared `utils.ts` in transport folder ?

File: packages/sources/elwood/src/transport/lwba.ts [hunk 4]
```diff
@@ -0,0 +1,165 @@
...
+
+export type ResponseMessage = PriceResponse | HeartbeatResponse
+
+export type WsTransportTypes = BaseEndpointTypesLwba & {
+  Provider: {
+    WsMessage: ResponseMessage
+  }
+}
+
+export const transport: WebSocketTransport<WsTransportTypes> =
```
Comment (karen-stepanyan) [inline packages/sources/elwood/src/transport/lwba.ts:49]: How is this different from a transport used in `crypto` endpoint ? Is it only the top level `result` response that is null in this transport ? Do we want to do this to fix the alert issues with missing/wrong ask/bid/mid prices ? If so can me have the logic in one place and reuse it somehow ?

File: packages/sources/gsr/src/adapter.ts [hunk 5]
```diff
@@ -0,0 +1,143 @@
...
+    if (!includesMap[from]) {
+      includesMap[from] = {}
+    }
+    includesMap[from][to] = includes[0]
+  }
+
+  return includesMap
+}
+
+export class GSRPriceAdapter<
```
[the remark this answers is not in the document]
Comment (mjk90) [reply packages/sources/gsr/src/adapter.ts:50]: You're right, we can just extend from PriceAdapter. For some reason, I had the idea that we couldn't override validateOutput using PriceAdapter but it looks like we can. I'll update this 🙏

File: packages/sources/elwood/src/endpoint/crypto.ts (hunk 1 above)
Comment (mjk90) [reply packages/sources/elwood/src/endpoint/crypto.ts:20]: Good point. I don't think it's necessary to actually make this change so I'll revert the response to it's original format

**Review by mxiao-cll — APPROVED** [review]

**Pull request merged**